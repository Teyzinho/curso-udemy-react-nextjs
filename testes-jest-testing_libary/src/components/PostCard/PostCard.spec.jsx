import { render, screen } from "@testing-library/react";

import { PostCard } from ".";

  //mock de props
const props = {
  title: "Titulo",
  body: "Body",
  id: 1,
  cover: "img/img.png",
};

describe("<PostCard/>", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: props.title }))
      .toHaveAttribute("src", props.cover);
    expect(screen.getByRole('heading',{name: /Titulo 1/i})).toBeInTheDocument()
    expect(screen.getByText(props.body)).toBeInTheDocument()
  });

  it('should match snapshot', () => { //Verifica se o componente foi editado ou n√£o a partir de uma snapshot ('foto')
    const {container} = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot()
  }) 
});
