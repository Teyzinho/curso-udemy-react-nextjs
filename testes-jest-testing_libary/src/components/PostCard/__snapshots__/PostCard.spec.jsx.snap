// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Titulo"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Titulo
       
      1
    </h2>
    <p>
      Body
    </p>
  </div>
</div>
`;
